src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -523,22 +523,24 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
             return document;
         }
 
-        private static BlittableJsonReaderObject ChangeSnapshotFlag(DocumentsOperationContext context, BlittableJsonReaderObject document, string documentId, string snapshotFlag, string flag)
+        private static BlittableJsonReaderObject RevertSnapshotFlags(DocumentsOperationContext context, BlittableJsonReaderObject document, string documentId)
         {
-            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) &&
-                metadata.TryGet(snapshotFlag, out BlittableJsonReaderObject bjro))
+            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) == false)
+                return document;
+
+            if (metadata.TryGet(Constants.Documents.Metadata.RevisionCounters, out BlittableJsonReaderObject bjro))
             {
                 var names = bjro.GetPropertyNames();
 
                 metadata.Modifications = new DynamicJsonValue(metadata);
-                metadata.Modifications.Remove(snapshotFlag);
+                metadata.Modifications.Remove(Constants.Documents.Metadata.RevisionCounters);
                 var arr = new DynamicJsonArray();
                 foreach (var name in names)
                 {
                     arr.Add(name);
                 }
 
-                metadata.Modifications[flag] = arr;
+                metadata.Modifications[Constants.Documents.Metadata.Counters] = arr;
                 document.Modifications ??= new DynamicJsonValue();
                 document.Modifications[Constants.Documents.Metadata.Key] = metadata;
```

Comment:
Let's make a function out of this `RevertCountersSnapshotFlag` which will return `true` or `false` and do the `context.ReadObject( ... )`  only this function returned `true`

Reply:
so for 5.2 we will simply add more to the modification if we need to change both time-series and coutners and do the `ReadObject` only once

Reply:
done